### ISSUE TITLE
Falco v0.17.1 build fails on rhel/centos

### ISSUE THREAD

**Issue by prankkelkar** [falcosecurity/falco#870]
**What happened**:
Trying to build falco from source on rhel/centos. Getting following error :
```
[  8%] Built target curl
make -f CMakeFiles/yamlcpp.dir/build.make CMakeFiles/yamlcpp.dir/depend
make[2]: Entering directory `/root/pk/palin/falco/build/release'
cd /root/pk/palin/falco/build/release && /usr/local/bin/cmake -E cmake_depends "Unix Makefiles" /root/pk/palin/falco /root/pk/palin/falco /root/pk/palin/falco/build/release /root/pk/palin/falco/build/release /root/pk/palin/falco/build/release/CMakeFiles/yamlcpp.dir/DependInfo.cmake --color=
Scanning dependencies of target yamlcpp
make[2]: Leaving directory `/root/pk/palin/falco/build/release'
make -f CMakeFiles/yamlcpp.dir/build.make CMakeFiles/yamlcpp.dir/build
make[2]: Entering directory `/root/pk/palin/falco/build/release'
[  8%] Creating directories for 'yamlcpp'
/usr/local/bin/cmake -E make_directory /root/pk/palin/falco/build/release/yamlcpp-prefix/src/yamlcpp
/usr/local/bin/cmake -E make_directory /root/pk/palin/falco/build/release/yamlcpp-prefix/src/yamlcpp
/usr/local/bin/cmake -E make_directory /root/pk/palin/falco/build/release/yamlcpp-prefix
/usr/local/bin/cmake -E make_directory /root/pk/palin/falco/build/release/yamlcpp-prefix/tmp
/usr/local/bin/cmake -E make_directory /root/pk/palin/falco/build/release/yamlcpp-prefix/src/yamlcpp-stamp
/usr/local/bin/cmake -E make_directory /root/pk/palin/falco/build/release/yamlcpp-prefix/src
/usr/local/bin/cmake -E touch /root/pk/palin/falco/build/release/yamlcpp-prefix/src/yamlcpp-stamp/yamlcpp-mkdir
[  9%] Performing download step (download, verify and extract) for 'yamlcpp'
cd /root/pk/palin/falco/build/release/yamlcpp-prefix/src && /usr/local/bin/cmake -P /root/pk/palin/falco/build/release/yamlcpp-prefix/src/yamlcpp-stamp/download-yamlcpp.cmake
-- Downloading...
   dst='/root/pk/palin/falco/build/release/yamlcpp-prefix/src/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
   timeout='none'
-- Using src='https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
-- Retrying...
-- Using src='https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
-- Retry after 5 seconds (attempt #2) ...
-- Using src='https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
-- Retry after 5 seconds (attempt #3) ...
-- Using src='https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
-- Retry after 15 seconds (attempt #4) ...
-- Using src='https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
-- Retry after 60 seconds (attempt #5) ...
-- Using src='https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz'
CMake Error at yamlcpp-stamp/download-yamlcpp.cmake:157 (message):
  Each download failed!

    error: downloading 'https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz' failed
         status_code: 1
         status_string: "Unsupported protocol"
         log:
         --- LOG BEGIN ---
         Protocol "https" not supported or disabled in libcurl

  Closing connection -1



         --- LOG END ---
         error: downloading 'https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz' failed
         status_code: 1
         status_string: "Unsupported protocol"
         log:
         --- LOG BEGIN ---
         Protocol "https" not supported or disabled in libcurl

  Closing connection -1



         --- LOG END ---
         error: downloading 'https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz' failed
         status_code: 1
         status_string: "Unsupported protocol"
         log:
         --- LOG BEGIN ---
         Protocol "https" not supported or disabled in libcurl

  Closing connection -1



         --- LOG END ---
         error: downloading 'https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz' failed
         status_code: 1
         status_string: "Unsupported protocol"
         log:
         --- LOG BEGIN ---
         Protocol "https" not supported or disabled in libcurl

  Closing connection -1



         --- LOG END ---
         error: downloading 'https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz' failed
         status_code: 1
         status_string: "Unsupported protocol"
         log:
         --- LOG BEGIN ---
         Protocol "https" not supported or disabled in libcurl

  Closing connection -1



         --- LOG END ---
         error: downloading 'https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz' failed
         status_code: 1
         status_string: "Unsupported protocol"
         log:
         --- LOG BEGIN ---
         Protocol "https" not supported or disabled in libcurl

  Closing connection -1



         --- LOG END ---




make[2]: *** [yamlcpp-prefix/src/yamlcpp-stamp/yamlcpp-download] Error 1
make[2]: Leaving directory `/root/pk/palin/falco/build/release'
make[1]: *** [CMakeFiles/yamlcpp.dir/all] Error 2
make[1]: Leaving directory `/root/pk/palin/falco/build/release'
make: *** [all] Error 2

```

**What you expected to happen**:
Successful completion of the build.

**How to reproduce it (as minimally and precisely as possible)**:
Follow steps to build from source given at https://falco.org/docs/source/

**Anything else we need to know?**:
Looks like error is related to curl's ssl support.

**Environment**:
- Falco version (use `falco --version`):
`0.17.1`
- System info 

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
```NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```
- Kernel (e.g. `uname -a`):
```Linux 5ba62e857d00 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux```
- Install tools (e.g. in kubernetes, rpm, deb, from source):
`Build from source`

- Other 
`Running centos inside docker container`

**Comment by fntlnz** [comment]
You don't have ssl support in libcurl.

```
Protocol "https" not supported or disabled in libcurl
```

**Comment by fntlnz** [comment]
See: https://curl.haxx.se/docs/faq.html#curl_1_SSL_is_disabled_https

**Comment by prankkelkar** [comment]
Yeah i had seen this already. anyways i am able to pull other https resources via curl on the vm. I suspect there is some issue with the bundled curl

**Comment by prankkelkar** [comment]
Infact i can even curl the url (https://s3.amazonaws.com/download.draios.com/dependencies/yaml-cpp-yaml-cpp-0.6.2.tar.gz) which is throwing error.

**Comment by guirish** [comment]
I am also encountering same error on RHEL 7.5. Any fix available to resolve it?

**Comment by prankkelkar** [comment]
Issue Solved !!! RHEL 7.x has lower version of cmake in the repository which was lower than required version by falco. So i had built cmake v3.7.2 from source. By default cmake is built without ssl support. Issue was resolved by building cmake v3.7.2 with ssl support.
